areas/mysql/src/AzureMcp.MySql/Services/MySqlService.cs

```diff
@@ -0,0 +1,482 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.RegularExpressions;
+using Azure;
+using Azure.Core;
+using Azure.ResourceManager.MySql.FlexibleServers;
+using AzureMcp.Core.Services.Azure;
+using AzureMcp.Core.Services.Azure.ResourceGroup;
+using AzureMcp.Core.Services.Azure.Tenant;
+using AzureMcp.MySql.Json;
+using Microsoft.Extensions.Logging;
+using MySqlConnector;
+
+namespace AzureMcp.MySql.Services;
+
+public class MySqlService(IResourceGroupService resourceGroupService, ITenantService tenantService, ILogger<MySqlService> logger) : BaseAzureService(tenantService), IMySqlService
+{
+    private readonly IResourceGroupService _resourceGroupService = resourceGroupService ?? throw new ArgumentNullException(nameof(resourceGroupService));
+    private readonly ILogger<MySqlService> _logger = logger;
+    private string? _cachedEntraIdAccessToken;
+    private DateTime _tokenExpiryTime;
+    private readonly object _tokenLock = new object();
+
+    // Static arrays for security validation - initialized once per class
+    private static readonly string[] DangerousKeywords =
```

Comment:
Can we do the same thing for regular SQL and put this in a shared lib that it can use?